src/sentry/api/event_search.py

```diff
@@ -10,7 +10,6 @@
 
 from django.utils.functional import cached_property
 from parsimonious.exceptions import IncompleteParseError
-from parsimonious.expressions import Optional
```

Comment:
`Optional` was removed in `0.10.0` in favor of a function that creates an optional `Quantifier`. For this reason, I had to adapt the logic to consider some text as optional if a `Node` has no matched `text`.